pkg/kubecost/allocation.go

```diff
@@ -282,36 +310,41 @@ func (parcs ProportionalAssetResourceCosts) Clone() ProportionalAssetResourceCos
 	return cloned
 }
 
-func (parcs ProportionalAssetResourceCosts) Insert(parc ProportionalAssetResourceCost, insertByNode bool) {
-	if !insertByNode {
-		parc.Node = ""
+func (parcs ProportionalAssetResourceCosts) Insert(parc ProportionalAssetResourceCost, insertByName bool) {
+	if !insertByName {
+		parc.Name = ""
+		parc.Type = ""
 		parc.ProviderID = ""
 	}
-	if curr, ok := parcs[parc.Key(insertByNode)]; ok {
+
+	if curr, ok := parcs[parc.Key(insertByName)]; ok {
 
 		toInsert := ProportionalAssetResourceCost{
-			Node:                curr.Node,
-			Cluster:             curr.Cluster,
-			ProviderID:          curr.ProviderID,
-			CPUTotalCost:        curr.CPUTotalCost + parc.CPUTotalCost,
-			CPUProportionalCost: curr.CPUProportionalCost + parc.CPUProportionalCost,
-			RAMTotalCost:        curr.RAMTotalCost + parc.RAMTotalCost,
-			RAMProportionalCost: curr.RAMProportionalCost + parc.RAMProportionalCost,
-			GPUProportionalCost: curr.GPUProportionalCost + parc.GPUProportionalCost,
-			GPUTotalCost:        curr.GPUTotalCost + parc.GPUTotalCost,
+			Name:                         curr.Name,
+			Type:                         curr.Type,
+			Cluster:                      curr.Cluster,
+			ProviderID:                   curr.ProviderID,
+			CPUTotalCost:                 curr.CPUTotalCost,
```

Comment:
It's tough - I'm trying to accomplish this without needing to know whether this is an accumulation or not. In the case this is not an accumulation, let's say we have have 2 containers getting aggregated into pods. Each container costs 1/10th of CPU. We would expect, then, that the resulting pod would cost 2/10ths of a CPU, right? The CPU total cost stays the same, so we should not need to add the CPU total costs together. 

This breaks down with accumulation. Each time slice has both separate overall CPU costs AND allocated CPU costs. Let's say our pod ran for 3 hours, consuming 2/10ths of the CPU for all 3 hours. After accumulation across all 3 hours, we would expect the pod to still have a PARC on the CPU of 2/10ths. This same logic then breaks down. if we keep this logic and only add the numerators, then at the end of the 3 hours, our parc says that we consumed 6/10ths of the CPU, which is not correct.

Reply:
Ah yes, definitely an issue. It's tough, keeping the total cost on the individual allocations, isn't it? It'd almost be better to somehow track the totals separately. So does this current solution work for both acc and agg? Or just fixed for agg, but now broken again for acc?

Reply:
Does the TotalsStore help you at all here? Are you familiar with that?

Reply:
https://github.com/opencost/opencost/blob/develop/pkg/kubecost/totals.go#L13-L38

Reply:
If you have access to that, you may be able to just record the "local" proportional costs and look up the totals at the end.